repair/repair.cc

```diff
@@ -1092,9 +1093,17 @@ int repair_service::do_repair_start(sstring keyspace, std::unordered_map<sstring
         options.column_families.size() ? options.column_families : list_column_families(db.local(), keyspace);
     if (cfs.empty()) {
         rlogger.info("repair[{}]: completed successfully: no tables to repair", id.uuid);
-        return id.id;
+        co_return id.id;
     }
 
+    db::event_params params = {
+        {"keyspace_name", keyspace},
+        {"options", format("{}", options_map)},
+        {"id", format("{}", id.id)},
+        {"uuid", format("{}", id.uuid)},
+    };
```

Comment:
Items I think we'd better add:

reason - reason for the repair (can be plain repair or node operations)
ranges - ranges requested to repair
tables - tables requested to repair

Currently, this patch ignores repair for repair based node operations (repair_service::do_sync_data_using_repair).  If we move the event into 
repair_tracker::run(), we can cover both repair and node ops automatically.

We do not have "options" for rbno repair. But repair options passed by user is not very useful,  perhaps we can ignore it even for regular repair.

Reply:
> But repair options passed by user is not very useful, perhaps we can ignore it even for regular repair.

Why is not useful, @asias ? What "options" are we talking about here? Does include DCs whitelisting, for example? I think we every single parameter given to a repair task is important - even if you think it's not important to you now your opinion may change in the future.

Reply:
I am talking about the options passed by the api. It is more useful to record the options after it is parsed. For example user may pass ranges, startToken, endToken, primaryRange to specify the actual ranges to repair. The ranges can even be empty which means all the ranges of the node. After parsing the options, we know which ranges the repair will repair no matter how the user specify it. Similarly, we have dataCenters / hosts / ignore_nodes to control which nodes to repair. If we just record the address of the nodes repair actually talks with, it is more informative. People who reads the table does not need to parse the options again.